calculator.c

```diff
@@ -0,0 +1,109 @@
+#include<stdio.h>
+#include<string.h>
+#include <stdbool.h>
+bool error=false;//to check if nay error encountered
+bool isOperator(char character){//function to check for an operator
+    if(character=='+' || character=='*' || character=='/' || character=='-'){
+        return true;
+    }
+    return false;
+}
+int precedence(char operator){//to check precedence of the operator
+    if(operator=='/')
+        return 4;
+    else if(operator=='*')
+        return 3;
+    else if(operator=='+')
+        return 2;
+    else
+        return 1;
+}
+int calculate(int value1,int value2,char operator){//to calculate the expression
+    int tempResult=0;
+    if(operator=='+'){
+        tempResult=value1+value2;
+    }
+    else if(operator=='-'){
+        tempResult=value1-value2;
+    }
+    else if(operator=='/'){
+        if(value2==0){
+            printf("Error: Division by zero.\n");
+            error=true;
+            return 0;
+        }
+        tempResult=value1/value2;
+    }
+    else if(operator=='*'){
+        tempResult=value1*value2;
+    }
+    return tempResult;
+}
+
+int evaluate(char expression[100]){//to evaluate complete expression
```

Comment:
It seems your code isn't handling double operators correctly. For example:
Input: 3++5
Expected Output: Error: Invalid string